src/functions/meme/main.py

```diff
@@ -0,0 +1,54 @@
+import json
+import os
+from typing import Optional, Tuple
+
+import functions_framework
+import requests
+from flask.wrappers import Request
+
+MEME_FILE = "memes.json"
+
+NO_EMOJI = "https://cdn.discordapp.com/emojis/823403768448155648.webp"
+USAGE_GUIDE = "Usage: !meme <Meme name> <Top text> <Bottom text (Optional)>"
+UNSUPORTED_MEME = "Meme not in supported list."
+
+
+@functions_framework.http
+def meme(request: Request) -> Tuple[str, int]:
+    """Return url to generated meme."""
+
+    request_json: Optional[dict] = request.get_json(silent=True)
+
+    with open(MEME_FILE, "r", encoding="utf-8") as meme_file:
+        memes: dict = json.load(meme_file)
+
+    if request_json:
+        params: list = request_json.get("params", [""])
+
+        if len(params) != 2 and len(params) != 3:
+            return USAGE_GUIDE, 200
```

Comment:
we might want to send the usage guide as an Ephemeral message, with how complex is it to remember the meme name, this will create a lot of spam

Reply:
Ephemeral messages are only for Slash Commands ([Source](https://support.discord.com/hc/fr/articles/1500000580222-Ephemeral-Messages-FAQ)).